src/components/Form/index.js

```diff
@@ -0,0 +1,184 @@
+         }
+      })
+   }
+
+   componentWillReceiveProps(props) {
+      let initialValues = props.initialValues
+
+      if (initialValues && !this.initialValuesLoaded) {
+         const newState = {}
+
+         initialValues = dot.dot(initialValues)
+
+         props.children.forEach((child) => {
+            if (this.childIsRelevant(child)) {
+               const value = initialValues
+                  ? initialValues[child.props.name] || ''
+                  : ''
+               newState[child.props.name] = {
+                  value,
+                  meta: {
+                     touched: false,
+                     error: this.checkError(child, value, props)
+                  }
+               }
+            }
+         })
+
+         this.initialValuesLoaded = true
+         this.setState(newState)
+      }
+   }
+
+   getTestFromString(str) {
+      switch (str) {
+         case 'required':
+            return value => !!value
+         case 'email':
+            return (value) => {
+               const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ // eslint-disable-line
+               return regex.test(value)
+            }
+         default:
+            break
+      }
+   }
+
+   checkError(child, initialValue, initialProps) {
+      const value = initialValue && typeof initialValue === 'string'
+         ? initialValue
+         : (this.state[child.props.name] && this.state[child.props.name].value) || ''
+      const props = initialProps || this.props
+
+      const validations = props.validations && props.validations[child.props.name]
+      if (!validations) return ''
+
+      for (let i = 0; i < validations.length; i += 1) {
+         const validation = validations[i]
+         const test = typeof validation.test === 'string'
+            ? this.getTestFromString(validation.test)
+            : validation.test
+         if (!test(value)) {
+            return validation.message
+         }
+      }
+
+      return ''
+   }
+
+   childIsRelevant(child) {
+      const types = ['input', 'textarea']
+      return types.indexOf(child.type.type) !== -1
+   }
+
+   handleBlur(event, child) {
+      const newState = {}
+      newState[child.props.name] = {
+         ...this.state[child.props.name],
+         meta: {
+            error: this.checkError(child),
+            touched: true
+         }
+      }
+
+      this.setState(newState)
+   }
+
+   handleChange(event, child) {
+      const value = event.target.value
+
+      const newState = {}
+      newState[child.props.name] = {
+         value,
+         meta: {
+            error: this.checkError(child, value),
+            touched: true
+         }
+      }
+
+      this.setState(newState)
+   }
+
+   handleSubmit(event) {
+      event.preventDefault()
+
+      const data = {}
+      const errors = {}
+      Object.keys(this.state).forEach((key) => {
```

Comment:
Was this desired functionality? I've already implemented it so that if a non-relevant initialValue is set, it won't appear as a property of the `data` object in the `onSubmit` prop. If external logic needs to be implemented to add extra properties to `data` before being sent off to the API, I think it should be done in `onSubmit` since it's not `<Form>`'s job to deal with that. Thoughts?